rfcs/function-table-clone.md

```diff
@@ -0,0 +1,64 @@
+# table.clone
+
+## Summary
+
+Add `table.clone` function that, given a table, produces a copy of that table with the same keys/values/metatable.
+
+## Motivation
+
+There are multiple cases today when cloning tables is a useful operation.
+
+- When working with tables as data containers, some algorithms may require modifying the table that can't be done in place for some reason.
+- When working with tables as objects, it can be useful to obtain an identical copy of the object for further modification, preserving the metatable.
+- When working with immutable data structures, any modification needs to clone some parts of the data structure to produce a new version of the object.
+
+While it's possible to implement this function in user code today, it's impossible to implement it with maximum efficiency; furthermore, cloning is a reasonably fundamental
+operation so from the ergonomics perspective it can be expected to be provided by the standard library.
+
+## Design
+
+`table.clone(t)` takes a table, `t`, and returns a new table that:
+
+- has the same metatable
+- has the same keys and values
+- is not frozen, even if `t` was
+
+The copy is shallow: implementing a deep recursive copy automatically is challenging (for similar reasons why we decided to avoid this in `table.freeze`), and often only certain keys need to be cloned recursively which can be done after the initial clone.
+
+The table can be modified after cloning; as such, a primitive like `Object.assign` could be built on top of this.
```

Comment:
Can you note the programming language with `Object.assign` that is referenced here?

Reply:
I now see that it's referenced in a later section, maybe move that up here.

Reply:
Removed this instead